### PR TITLE
fix: don't trigger chain switch if all actions are done from source chain [LF-1131]

### PR DESCRIPTION
**Pull request by Abhikumar98**
# FIX

Switching chains shouldn't be triggered if all actions are done from source chain [LF-1131]

## TL;DR
Disable chain switching if all required actions are already completed on the source chain.

## DESCRIPTION
Currently, the chain switching window is triggered even if all required actions are already completed on the source chain. This can be an unnecessary interruption for users who only need to wait for the tokens to be received in the receiving chain.

This feature will disable the chain switching window if all required actions are already completed on the source chain, and the user only needs to wait for the tokens to be received in the receiving chain.

## STEPS TO TEST
1. Bridge USDC from Ethereum to Polygon.
2. Switch your chains to a third network, say Binance.
3. Upon starting the process, the chain would be promptly switched to the source chain which is Ethereum.
4. Once the allowance is given and this transaction is signed, all the actions that are required from the source are completed, and now we will have to wait for the tokens to be deposited in the receiving chain.
5. While the screen shows "waiting for the funds to receiving receiving", go ahead and switch chains in Metamask and reload the page.

## EXPECTED RESULTS
Since the source chain actions are all done, there is no need for any further chain switching unless there is a swap to be carried out as a step.

## CHECKLIST
- [x] I have performed a self-review of my code
- [ ] I have commented the code I wrote to make sure the next developers can understand it
- [ ] I have added tests that cover the new functionality
- [x] I have made sure that existing tests are still passing

[LF-1131]: https://lifi.atlassian.net/browse/LF-1131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Addminus — CHANGES_REQUESTED** [review]

File: src/execution/StepExecutor.ts [hunk 1]
```diff
@@ -54,6 +54,15 @@ export class StepExecutor {
 
   executeStep = async (signer: Signer, step: LifiStep): Promise<LifiStep> => {
     // Make sure that the chain is still correct
+
+    const recievingChainProcess = step.execution?.process.find(
+      (process) => process.type === 'RECEIVING_CHAIN'
+    )
+
+    if (recievingChainProcess?.substatus === 'WAIT_DESTINATION_TRANSACTION') {
+      return step
```
[the remark this answers is not in the document]
Comment (Addminus) [reply src/execution/StepExecutor.ts:67]: Have you checked if the transaction ever finishes successfully?

[the remark this answers is not in the document]
Comment (Abhikumar98) [reply src/execution/StepExecutor.ts:67]: ahh got it.
I well it finished but the finished transaction wasn't the reloaded page.

I see the mistake, I'm returning before the step execution starts. Fixing it!

[the remark this answers is not in the document]
Comment (Abhikumar98) [reply src/execution/StepExecutor.ts]: @Addminus I've fixed this. Please check it out once again.

**Review by Addminus — APPROVED** [review]

**Comment by chybisov** [comment]
@Abhikumar98 please add tests for this :)
